bloom-housing/bloom
# feat: listings management draft and publish validation backend & frontend

# Pull Request Template

## Issue

Addresses #1543 #1541 #1811 #1542


## Description

Adds the backend and frontend pieces for validations on listings management on both draft and publish. I also refactored a little how we are submitting data so we don't need to keep track of status or submitData in state.

For draft, required fields are:
* Listing name
* Jurisdiction (only shows if user has more than one, otherwise defaults)

For publish, required fields are:
* Listing name
* Jurisdiction
* Developer
* Image
* Is there a digital application?
* Is there a paper application? 
* Is this a referral opportunity? 
* Building Address (specifically street, city, state, zipCode)
* Units - At least one unit
* Deposit Min (has a default)
* Deposit Max (has a default)
* Rental Assistance (has a default)
* How is the application review order determined? (has a default)
* Is the waitlist open?
* Leasing agent email
* Leasing agent name
* Leasing agent phone


## Type of change


## How Can This Be Tested/Reviewed?

Create a new listing from scratch and without filling anything out and try to save as draft and publish and the frontend validations should kick in. Also try to edit an existing listing.

The one kinda wonky frontend validation is buildingAddress, because if you send nothing we get a generic "buildingAddress" error but if at least one field is filled out we get errors specific to the four required fields. Should show as individual errors tho.

## Checklist:
